src/server/reducer.js

```diff
@@ -5,10 +5,42 @@ function reducer(state = {}, action) {
     case 'init_state':
       return core.initState(action.payload);
     case 'vote':
-      return core.vote(state, action.payload, action.clientId);
+      returnSet = core.vote(state, action.payload, action.clientId);
```

Comment:
We don't need to make any changes here. _Vote_ fn should be enough to creating new state.